### ISSUE TITLE
Outlier detection and remove feature interferes with the calculation of other currency

### ISSUE THREAD

**Issue by Reasonably** [opencost/opencost#1929]
**Describe the bug**
I'm using opencost through azure provider, which reads and applies data from pricesheet api asynchronously.   
Until it get the data from the pricesheet, it use the rate card api, and the currency is dollar ($). Since the currency of the pricesheet is KRW (₩), it looks like the price of the node has increased by about 1000 times after reading the data. So it exceeds the threshold of 30 times for outlier detection and the new information is not available.

**To Reproduce**
1. set azure provider
2. set billing account id and offer id

**Expected behavior**
after pricesheet downloaded, it can calculate node cost with KRW.

**Which version of OpenCost are you using?**
This may be the Kubecost release.
`quay.io/kubecost1/kubecost-cost-model:prod-1.103.1`

**Comment by mattray** [comment]
When the PR for https://github.com/opencost/opencost/issues/1688 comes in you should be able to set the currency, but by my understanding your other concern is the initial pricing seems really off until the pricesheet data is applied. Would providing your own custom [azure.json](https://github.com/opencost/opencost/blob/develop/configs/azure.json) with lower prices solve the issue?

**Comment by Reasonably** [comment]
@mattray 

I think #1688 is just enable to show other currency.  right? I want to use KRW for all calculating.
In first, I use default config(no azure.json) and set [service-key.json](https://github.com/opencost/opencost/blob/325ad0d014c5099a27b2c838d00ff00f49f377f7/pkg/cloud/models/models.go#L23). So it use ratecard api that its argument is USD  and US. In this case, there is the problem about detecting outlier I said.  
if there is no outlier detect feature, It will calculate using KRW after pricesheet download completed.  


So I added the config `azure.json` like below.
```
{
  "currencyCode": "KRW",
  "AzureBillingRegion": "KR"
}
```
I was hoping to use KRW to request the ratecard api.  I saw a log `INF Using ratecard query OfferDurableId eq 'MS-AZR-0003p' and Currency eq 'krw' and Locale eq 'en-US' and RegionInfo eq 'KR'`.  and I thought it work, but there was error after a few moments. 
```
2023-05-19T06:25:05.706984724Z INF Using ratecard query OfferDurableId eq 'MS-AZR-0003p' and Currency eq 'krw' and Locale eq 'en-US' and RegionInfo eq 'KR'
2023-05-19T06:26:34.029905135Z INF Init: AggregateCostModel cache warming enabled
2023-05-19T06:26:34.030043036Z INF EXPORT_CSV_FILE is not set, skipping CSV exporter
2023-05-19T06:26:34.030259238Z INF aggregation: cache warming defaults: 1d::::::::namespace::::weighted:false:false:true
2023-05-19T06:26:34.03042104Z INF ComputeAggregateCostModel: missed cache: 1d:1m:1.000000h:false (found false, disableAggregateCostModelCache true, noCache false)
2023-05-19T06:26:34.216371809Z WRN Could not parse default cpu price
2023-05-19T06:26:34.216417009Z WRN Could not parse default ram price
2023-05-19T06:26:34.216426109Z WRN cpuToRAMRatio[defaultCPU: 0.000000 / defaultRAM: 0.000000] is NaN. Setting to 0.
2023-05-19T06:26:34.216446909Z WRN Could not parse default cpu price
2023-05-19T06:26:34.216458409Z WRN Could not parse default ram price
```

I think it might be caused by not setting default CPU and MEM values in the azure.json, but I haven't tested it yet. Anyway, it seems right that the outlier detection feature is behaving unintentionally for intentional drastic changes. I think it would be good to improve or remove this logic, what do you think?

**Comment by mattray** [comment]
I'll escalate this to Kubecost engineering, it feels like something they should deal with at some point.

**Comment by Reasonably** [comment]
@mattray Over the course of Friday, I did some more testing. First of all, the reason for the parse error was a prometheus setup issue: it wasn't finding the node properly, so it was trying to write the data it found in the azure.json instead of using the data it got from the ratecard api. However, I hadn't put any default values in the azure.json, which was why the error was occurring.
After fixing these settings, I was able to see the data being applied just fine, but I still think there should be a way to modify opencost's outlier detect feature or apply the exchange rate.
